submitter/relayer/relayer.go

```diff
@@ -290,34 +331,66 @@ func (rl *Relayer) convertCkptToTwoTxAndSubmit(ckpt *ckpttypes.RawCheckpointResp
 	}, nil
 }
 
-// ChainTwoTxAndSend builds two chaining txs with the given data:
-// the second tx consumes the output of the first tx
-func (rl *Relayer) ChainTwoTxAndSend(data1 []byte, data2 []byte) (*types.BtcTxInfo, *types.BtcTxInfo, error) {
-	// recipient is a change address that all the
-	// remaining balance of the utxo is sent to
-	tx1, err := rl.buildTxWithData(data1, nil)
+func (rl *Relayer) retrySendTx2(ckpt *ckpttypes.RawCheckpointResponse) (*types.CheckpointInfo, error) {
+	_, data2, err := rl.encodeCheckpointData(ckpt)
+	if err != nil {
+		return nil, err
+	}
+
+	tx1 := rl.lastSubmittedCheckpoint.Tx1
+	if tx1 == nil {
```

Comment:
better to have a doc string for this function saying that the tx1 should not be nil

Reply:
But this function uses tx1 implicitly through state, I'll add the doc string above, but I would live the check in

Reply:
yep yep, keeping the check is good here